worldedit-core/src/main/java/com/sk89q/worldedit/EditSession.java

```diff
@@ -1478,29 +1482,33 @@ public int makeSphere(Vector pos, Pattern block, double radiusX, double radiusY,
                         }
                     }
 
-                    if (setBlock(pos.add(x, y, z), block)) {
-                        ++affected;
-                    }
-                    if (setBlock(pos.add(-x, y, z), block)) {
-                        ++affected;
-                    }
-                    if (setBlock(pos.add(x, -y, z), block)) {
-                        ++affected;
-                    }
-                    if (setBlock(pos.add(x, y, -z), block)) {
-                        ++affected;
-                    }
-                    if (setBlock(pos.add(-x, -y, z), block)) {
-                        ++affected;
-                    }
-                    if (setBlock(pos.add(x, -y, -z), block)) {
-                        ++affected;
-                    }
-                    if (setBlock(pos.add(-x, y, -z), block)) {
-                        ++affected;
+                    if(!hemi || (hemi && !upsideDown)) {
```

Comment:
This can be simplified to (!hemi || !upsideDown) but maybe this way it's more readable. (And possibly Java optimizes it out anyway.)